src/Microsoft.Dnx.Runtime.CommandParsing.Sources/CommandGrammar.cs

```diff
@@ -48,13 +48,13 @@ private CommandGrammar(Func<string, string> variable, bool preserveSurroundingQu
 
         public static string[] Process(string text, Func<string, string> variables, bool preserveSurroundingQuotes)
         {
-            var grammer = new CommandGrammar(variables, preserveSurroundingQuotes);
+            var grammar = new CommandGrammar(variables, preserveSurroundingQuotes);
             var cursor = new Cursor(text, 0, text.Length);
 
-            var result = grammer.Parse(cursor);
+            var result = grammar.Parse(cursor);
             if (!result.Remainder.IsEnd)
             {
-                throw new Exception("TODO: malformed command text");
+                throw new ArgumentException($"Malformed command text '{text}'");
```

Comment:
Pass the parameter name to the constructor as well (there's an overload that takes it)